app/components/activity_list_component.html.erb

```diff
@@ -0,0 +1,6 @@
+<ul class="op-activity-list">
+  <% @events.each do |event| -%>
+    <% journal = @journals_by_id[event.event_id] -%>
+    <%= render(ActivityItemComponent.new(event:, journal:, display_user: @display_user)) %>
+  <% end -%>
```

Comment:
Nitpicking as I am admittedly picky right now since I want to understand the patterns to use. Those 4 lines are all pure ruby code, no tag at all. Since there is an underlying ruby class now, that is one of the benefits of `ViewComponent` I would have expected to have as much non template code in the `component.rb` class as possible. 

So I would at least have preferred something like: 

```
<% journals.each do |journal| %>
   <%= render(ActivityItemComponent.new(event:, journal:, display_user: @display_user)) %>
<% end %>
```

Where `journals` would be the collection of all journals associated to the `@events`. That would also ease the transition to only working with journals as well.

But I am wondering whether I would even prefer:

```
   <%= render_journals %>
```

since there are no tags. Today, I would prefer it for the reason that I find a mix of languages harder to read. Tomorrow, it might be different, though.

Reply:
Not sure if I addressed this one with my last commit. 

Now it's
```
  <% @events.each do |event| -%>
    <%= render(Activities::ItemComponent.new(event:, display_user: @display_user)) %>
  <% end -%>
```

You would like to have it put into the `.rb` file?

Reply:
I found another way to do it:

```
<%= render(Activities::ItemComponent.with_collection(@events, display_user: @display_user)) %>
```

[It handles the iteration + rendering part of it](https://viewcomponent.org/guide/collections.html).